streams/src/main/java/org/apache/kafka/streams/processor/internals/Tasks.java

```diff
@@ -270,6 +278,23 @@ Task task(final TaskId taskId) {
         return readOnlyActiveTasks;
     }
 
+    List<Task> orderedActiveTasks() {
+        return Collections.unmodifiableList(orderedActiveTasks);
+    }
+
+    void moveActiveTasksToTailFor(final String topologyName) {
```

Comment:
I am not sure we want to move all the tasks in a topology? Maybe we can do that by task or sub topology?

maybe topology is best but I will need to think about it a bit

Reply:
I told Bruno to do this -- we can discuss next week if you still have questions, but the motivation is to avoid letting the tasks of a topology/query get way out of sync with each other. This can result in missed output for example when a processor upstream of one side of a join is paused for a long time while the other continues proccessing records and trying to join them